thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js

```diff
@@ -3,31 +3,40 @@
  * @module manage/alert
  * @exports manage/alert
  */
-import Ember from 'ember';
+import Controller from '@ember/controller';
 
-export default Ember.Controller.extend({
+export default Controller.extend({
   /**
    * Set up to receive prompt to trigger page mode change.
-   * When replay is received as true, it indicates that this is a
-   * newly created alert and replay is needed in order to display anomaly data.
+   * When replay id is received it indicates that we need to check replay status
+   * before displaying alert function performance data.
    */
-  queryParams: ['replay', 'replayId'],
-  replay: false,
+  queryParams: ['replay'],
   replayId: null,
 
-  /**
-   * Actions for alert page
-   */
   actions: {
 
     /**
      * Placeholder for subscribe button click action
-     * @method onClickAlertSubscribe
-     * @return {undefined}
      */
     onClickAlertSubscribe() {
       // TODO: Set user as watcher for this alert when API ready
-    }
+    },
 
+    /**
+     * Handle navigation to edit route
+     */
+    onClickEdit() {
+      this.set('isOverViewModeActive', false);
+      this.set('isEditModeActive', true);
+      this.transitionToRoute('manage.alert.edit', this.get('id'));
+    },
+
+    /**
+     * Navigate back to manage base route
+     */
+    onClickNavToOverview() {
```

Comment:
Seems like there are a lot of actions on this controller, wondering if that's necessary? Ember best practices wants us to minimize controller usage whenever possible so just curious.

Reply:
These actions are needed for the links in the parent 'manage.alert' view.